### ISSUE TITLE
ReferenceError: htmlWebpackPlugin is not defined

### ISSUE THREAD

**Issue by trusktr** [jantimon/html-webpack-plugin#1357]
## Expected behaviour

? There's no docs.

## Current behaviour

Error

### Config

Copy the minimal `webpack.config.js` to produce this issue:

```js
    new HtmlWebpackPlugin({
      // for example, see https://github.com/jantimon/html-webpack-plugin/blob/master/examples/template-parameters/webpack.config.js
      templateParameters(_compilation, files, tags, options) {
        return {
          title: 'My App',
          tags,
          files,
          options,
        }
      },
    }),
```

Copy your template file if it is part of this issue:

```ejs
<%= title %>
```

## Additional context

I bet this error is because I'm not writing

```js
    new HtmlWebpackPlugin({
      templateParameters: (compilation, assets, assetTags, options) => {
        return {
          compilation,
          webpackConfig: compilation.options,
          htmlWebpackPlugin: {
            tags: assetTags,
            files: assets,
            options
          },
          'foo': 'bar'
        };
      },
    })
```

But there's hardly any documentation on the template feature. How would anyone know that the object needs a specific format?

**Comment by trusktr** [comment]
I see. The docs were confusing.

The description for the `template` option says

```
`webpack` relative or absolute path to the template. By default it will use `src/index.ejs` if it exists. Please see the [docs](https://github.com/jantimon/html-webpack-plugin/blob/master/docs/template-option.md) for details.
```

It isn't clear what "`webpack` relative or absolute path to the template" means. Then it says "By default it will use `src/index.ejs` if it exists", so I created a `src/index.ejs` file and finally when I ran webpack I got the error

```
ReferenceError: htmlWebpackPlugin is not defined
```

This is because it was not in fact loading the default template `src/index.ejs`, and loaded the internal default template.

Then I added `template: './index.ejs'` to HtmlWebpackPlugin, and using my original `templateParameters` function then worked as expected.

All in all, the docs could be more helpful.
